app/src/main/java/com/hedvig/app/feature/offer/ui/OfferActivity.kt

```diff
@@ -286,8 +287,9 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
         }
     }
 
-    private fun setSignState(signMethod: SignMethod) {
-        binding.signButton.bindWithSignMethod(signMethod)
+    private fun setSignButtonState(signMethod: SignMethod, checkoutTextRes: Int?) {
+        binding.signButton.text = checkoutTextRes?.let(::getString) ?: ""
+        binding.signButton.icon = signMethod.checkoutIconRes()?.let(::compatDrawable)
         binding.signButton.setHapticClickListener {
```

Comment:
A bit unrelated to this PR. But I've seen we use haptic click listener quite a lot. In compose we don't quite have a simple `. setHapticClickListener`, we should look into this especially now that we have our first buttons in Compose too.

Reply:
Very good point. Let's talk more about this with designers.